### ISSUE TITLE
Generated ampli client fails type checking for events with only optional properties

### ISSUE THREAD

**Issue by georgewitteman** [amplitude/Amplitude-TypeScript#347]
## Expected Behavior
The client generated from the `ampli` CLI should not produce any type errors for events with only optional properties.

## Current Behavior
Events generated by `ampli` that only contain optional properties show the following TypeScript error
```
Class 'PlaceOrderButtonClicked' incorrectly implements interface 'BaseEvent'.
  Types of property 'event_properties' are incompatible.
    Type 'PlaceOrderButtonClickedProperties | undefined' is not assignable to type '{ [key: string]: any; }'.
      Type 'undefined' is not assignable to type '{ [key: string]: any; }'.ts(2420)
```

Example generated event
```typescript
export class PlaceOrderButtonClicked implements BaseEvent {
  event_type = 'Place Order Button Clicked';

  constructor(
    public event_properties?: PlaceOrderButtonClickedProperties,
  ) {
    this.event_properties = event_properties;
  }
}
```

## Possible Solution
The [`BaseEvent`](https://github.com/amplitude/Amplitude-TypeScript/blob/e6b19b58479cf131404047649647ce39c4835082/packages/analytics-types/src/base-event.ts#L4) class should have `event_properties` defined as `event_properties?: { [key: string]: any } | undefined;`

## Steps to Reproduce
1. Create an event with all optional properties.
2. Run `ampli pull` to generate the client
3. Run type checking with `tsc --noEmit` on the project

## Environment
- ampli Version: `@amplitude/ampli/1.31.2 linux-x64 node-v16.19.1`
- JS SDK Version: `1.9.1`
- Installation Method: yarn
- TypeScript version: `Version 4.9.3`
- tsconfig.json: https://github.com/tsconfig/bases/blob/main/bases/next-strictest.json
- Browser and Version: N/A

**Comment by georgewitteman** [comment]
Came across another issue with `exactOptionalPropertyTypes` enabled.

Let's say you have some object with the type `{ foo?: string }` and an event `Test Event` that takes `foo_property` as an _optional_ property as well as some other properties that may or may not be required.

If you try to do this:
```typescript
ampli.testEvent({ foo_property: obj.foo, other_a: "a", other_b: "b" });
```

You'll get the error: `Type 'undefined' is not assignable to type 'string'.ts(2379)`.

Instead, you have to do this:
```typescript
ampli.testEvent({
  other_a: "a",
  other_b: "b",
  ...(obj.foo ? { foo_property: obj.foo } : {}),
});
```

It works, but is very verbose and annoying to write. In this example, `foo_property` should be defined as `foo_property?: string | undefined` instead of just `foo_property?: string`.
